tests/h/views/api/groups_test.py

```diff
@@ -311,6 +311,143 @@ def group(self, factories):
         return factories.Group(authority='example.com')
 
 
+@pytest.mark.usefixtures('create',
+                         'CreateGroupAPISchema',
+                         'group_service',
+                         'group_update_service',
+                         'GroupContext',
+                         'GroupJSONPresenter',)
+class TestUpsertGroup(object):
+
+    def test_it_proxies_to_create_if_group_empty(self, pyramid_request, create):
+        empty_group_context = mock.Mock()
+        empty_group_context.group = None
```

Comment:
This should be `create_autospec()` (we should basically never construct bare `Mock()`s or `MagicMock()`s)